### PR TITLE
fix code format problems

### PR DESCRIPTION
**Pull request by liangdzou**
### Your checklist for this pull request

Please review the [guidelines for contributing](http://solidity.readthedocs.io/en/latest/contributing.html) to this repository.

Please also note that this project is released with a [Contributor Code of Conduct](CONDUCT.md). By participating in this project you agree to abide by its terms.

### Checklist
- [x] Code compiles correctly
- [x] All tests are passing
- [x] New tests have been created which fail without the change (if possible)
- [x] README / documentation was extended, if necessary
- [x] Changelog entry (if change is visible to the user)
- [x] Used meaningful commit messages

### Description

Find code format problems by command:
find . -name "*.h" | xargs egrep "\<if\>.*\{$" -nr | grep -v "./build" | grep -v "./test"
find . -name "*.cpp" | xargs egrep "\<if\>.*\{$" -nr | grep -v "./build" | grep -v "./test"
find . -name "*.h" | xargs egrep "\<for\>.*\{$" -nr | grep -v "./build" | grep -v "./test"
find . -name "*.cpp" | xargs egrep "\<for\>.*\{$" -nr | grep -v "./build" | grep -v "./test"

### CODE REVIEW

File: libdevcore/SHA3.cpp [hunk 1]
```diff
@@ -166,24 +166,25 @@ mkapply_sd(setout, dst[i] = src[i])  // setout
...
-  // Xor in the last block.
-  xorin(a, in, inlen);
-  // Apply P
-  P(a);
-  // Squeeze output.
-  foldP(out, outlen, setout);
-  setout(a, out, outlen);
-  memset(a, 0, 200);
-  return 0;
+	if ((out == NULL) || ((in == NULL) && inlen != 0) || (rate >= Plen))
```
Comment (axic) [inline libdevcore/SHA3.cpp]: I'm not sure we should spend time on this file, since it is an upstream code, but the rest looks good.

Comment (liangdzou) [reply libdevcore/SHA3.cpp]: ok, I will revert the change on this one. Thanks :-)

Comment (liangdzou) [reply libdevcore/SHA3.cpp]: reverted for the file, and squashed with the previous commit.

**Review by axic — APPROVED** [review]

**Comment by chriseth** [comment]
@liangdzou could you add these scripts to the existing "trailing whitespace" CI job (and rename it)?

**Comment by chriseth** [comment]
The checks could also find cases of `if(`

**Comment by liangdzou** [comment]
yeah, will do it later

**Pull request merged**